src/auth/token.ts

```diff
@@ -54,19 +53,31 @@ export function readAuthFromWebsite(): AuthData {
   };
 }
 
-export async function getExtensionToken(): Promise<string | null> {
-  const valueJSON = await readStorage(STORAGE_EXTENSION_KEY);
-  return valueJSON ? JSON.parse(valueJSON as string).token : undefined;
-}
+async function readAuthData(): Promise<AuthData | Partial<AuthData>> {
+  const storage = await browser.storage.local.get({
+    [STORAGE_EXTENSION_KEY]: {},
+  });
+  // eslint-disable-next-line security/detect-object-injection -- Local constant
+  const data = storage[STORAGE_EXTENSION_KEY];
```

Comment:
The storage API is awkward to use; We should probably keep the helpers _but_ update them to:

- allow non-strings
- accept defaults via `.get(key, defaultValue)`

In a future PR: Proposed API, one for `local` and one for `sync`:

```ts
interface Storage {
	get<T>(key: string, default?: unknown): Promise<T>;
	set(key: string, value: unknown): Promise<void>;
}
```

It'd change this to
```suggestion
  const data = await storage.local.get(STORAGE_EXTENSION_KEY, {});
```

Reply:
I agree it makes sense to move toward the sensible API for storage